### ISSUE TITLE
automated tests missing in 3.7.0

### ISSUE THREAD

**Issue by velle** [OSGeo/gdal#7771]
## Expected behavior and actual behavior.

The following ctest command was run from within the build folder, straight after "cmake --build ."

#### Observed behavior with version 3.5.0:
```
$ ctest
Test project /home/velle/c/my_gdal/gdal-3.5.0/build
    Start 1: test-fuzzer-gdal
1/5 Test #1: test-fuzzer-gdal ......................***Exception: SegFault  0.17 sec
    Start 2: test-fuzzer-ogr
2/5 Test #2: test-fuzzer-ogr .......................***Exception: SegFault  0.17 sec
    Start 3: test-fuzzer-osr_set_from_user_input
3/5 Test #3: test-fuzzer-osr_set_from_user_input ...***Exception: SegFault  0.18 sec
    Start 4: test-fuzzer-wkb_import
4/5 Test #4: test-fuzzer-wkb_import ................***Exception: SegFault  0.17 sec
    Start 5: test-fuzzer-wkt_import
5/5 Test #5: test-fuzzer-wkt_import ................***Exception: SegFault  0.17 sec

0% tests passed, 5 tests failed out of 5

Total Test time (real) =   0.87 sec

The following tests FAILED:
	  1 - test-fuzzer-gdal (SEGFAULT)
	  2 - test-fuzzer-ogr (SEGFAULT)
	  3 - test-fuzzer-osr_set_from_user_input (SEGFAULT)
	  4 - test-fuzzer-wkb_import (SEGFAULT)
	  5 - test-fuzzer-wkt_import (SEGFAULT)
Errors while running CTest
Output from these tests are in: /home/velle/c/my_gdal/gdal-3.5.0/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

#### Observed behavior, with version 3.7.0:
```
~/c/my_gdal/gdal-3.7.0/build$ ctest
Test project /home/velle/c/my_gdal/gdal-3.7.0/build
No tests were found!!!
```

## Steps to reproduce the problem.

```
cd ~/c/my_gdal/
wget http://download.osgeo.org/gdal/3.7.0/gdal-3.7.0.tar.gz
tar -xvzf gdal-3.7.0.tar.gz
cd gdal-3.7.0
mkdir build
cd build
cmake ..
cmake --build .
```

And then, from the build folder execute "ctest". 

```
~/c/my_gdal/gdal-3.7.0/build$ ctest
Test project /home/velle/c/my_gdal/gdal-3.7.0/build
No tests were found!!!
```

## Operating system

Ubuntu 22.04 64 bit

## GDAL version and provenance

GDAL 3.7.0.

**Comment by rouault** [comment]
Tests are removed from the official tarball to keep it small. If you want to run tests, the best is to use the snapshot of the tag at https://github.com/OSGeo/gdal/archive/refs/tags/v3.7.0.tar.gz

**Comment by velle** [comment]
That explains. Thanks :)

After having followed the "Building From Source" page (https://gdal.org/development/building_from_source.html), the next page is on testing and explains how to do that. So thats what I did. 

I spent some time reading the docs and trying to make those tests run before I realized that it probably was not possible to run them at all :) Of course it did not help that I had never tried ctest before. 

Perhaps the test page should start with a small note, that since version X, testing is only included in such and such downloads. After all, the previous page just explained how to get the source by using wget for the tarball :)
